src/main/java/com/avispl/symphony/dal/infrastructure/management/tripleplay/TriplePlayAggregatorCommunicator.java

```diff
@@ -0,0 +1,601 @@
+import com.avispl.symphony.dal.infrastructure.management.tripleplay.common.monitoring.HardwareMetric;
+import com.avispl.symphony.dal.infrastructure.management.tripleplay.common.monitoring.NetworkMetric;
+import com.avispl.symphony.dal.infrastructure.management.tripleplay.dto.Activity;
+import com.avispl.symphony.dal.infrastructure.management.tripleplay.dto.Client;
+import com.avispl.symphony.dal.infrastructure.management.tripleplay.dto.MonitoringData;
+import com.avispl.symphony.dal.infrastructure.management.tripleplay.dto.Service;
+import com.avispl.symphony.dal.infrastructure.management.tripleplay.dto.ServiceWrapper;
+import com.avispl.symphony.dal.infrastructure.management.tripleplay.dto.request.implement.ControllingRequest;
+import com.avispl.symphony.dal.infrastructure.management.tripleplay.dto.request.implement.GetAllServicesRequestBody;
+import com.avispl.symphony.dal.infrastructure.management.tripleplay.dto.request.implement.QueryClientRequestBodyV2;
+import com.avispl.symphony.dal.infrastructure.management.tripleplay.dto.request.implement.QueryClientsListRequestBody;
+
+import com.avispl.symphony.api.dal.control.Controller;
+import com.avispl.symphony.api.dal.dto.control.AdvancedControllableProperty;
+import com.avispl.symphony.api.dal.dto.control.ControllableProperty;
+import com.avispl.symphony.api.dal.dto.monitor.Statistics;
+import com.avispl.symphony.api.dal.dto.monitor.aggregator.AggregatedDevice;
+import com.avispl.symphony.api.dal.error.ResourceNotReachableException;
+import com.avispl.symphony.api.dal.monitor.Monitorable;
+import com.avispl.symphony.api.dal.monitor.aggregator.Aggregator;
+import com.avispl.symphony.dal.communicator.RestCommunicator;
+import com.avispl.symphony.dal.util.StringUtils;
+import com.avispl.symphony.dal.util.ControllablePropertyFactory;
+
+/**
+ * Communicator
+ *
+ * @author Harry / Symphony Dev Team<br>
+ * Created on 3/13/2023
+ * @since 1.0.0
+ */
+public class TriplePlayAggregatorCommunicator extends RestCommunicator implements Aggregator, Monitorable, Controller {
+
+	class ClientLoader implements Runnable {
+		private volatile List<String> clientIps;
+
+		public ClientLoader(List<String> clientIps) {
+			this.clientIps = clientIps;
+		}
+
+		@Override
+		public void run() {
+
+			if (!cachedClients.isEmpty()) {
+				retrieveClient(this.clientIps);
+			}
+		}
+	}
+
+	/**
+	 * ReentrantLock to prevent null pointer exception to localExtendedStatistics when controlProperty method is called before GetMultipleStatistics method.
+	 */
+	private final ReentrantLock reentrantLock = new ReentrantLock();
+
+	/**
+	 * Executor that runs all the async operations
+	 */
+	private static ExecutorService executorService;
+
+	private volatile int localPollingInterval = TriplePlayConstrant.MIN_POLLING_INTERVAL;
+	private volatile int deviceStatisticsCollectionThreads;
+
+	/**
+	 * store pollingInterval adapter properties
+	 */
+	private volatile String pollingInterval;
```

Comment:
provide the getter, setter for adapter properties pollingInterval

Reply:
I fixed it